ffi/README.md

```diff
@@ -1,13 +1,52 @@
-# A firewood golang interface
+# Firewood Golang FFI
 
-This allows calling into firewood from golang
+The FFI package provides a golang FFI layer for Firewood.
 
-## Building
+## Building Firewood Golang FFI
 
-First, build the release version (`cargo build --release`). This creates the ffi
-interface file "firewood.h" as a side effect.
+The Golang FFI layer uses a CGO directive to locate a C-API compatible binary built from Firewood. Firewood supports both seamless local development and a single-step compilation process for Go projects that depend or transitively depend on Firewood.
 
-Then, you can run the tests in go, using `go test .`
+To do this, [firewood.go](./firewood.go) includes CGO directives to include multiple search paths for the Firewood binary in the local `target/` build directory and `ffi/libs`. For the latter, [attach-static-libs](../.github/workflows/attach-static-libs.yaml) GitHub Action pushes an FFI package with static libraries attached for the following supported architectures:
+
+- x86_64-unknown-linux-gnu
+- aarch64-unknown-linux-gnu
+- aarch64-apple-darwin
+- x86_64-apple-darwin
+
+to a separate repo [firewood-go](https://github.com/ava-labs/firewood-go) (to avoid including binaries in the Firewood repo).
+
+### Local Development
+
+[firewood.go](./firewood.go) includes CGO directives to include builds in the `target/` directory.
+
+Firewood prioritizes builds in the following order:
+
+1. maxperf
```

Comment:
Is this correct? Seems like the other way is better (debug before release before maxperf)?

Reply:
I'll switch in a follow up. This is what you requested previously.